aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -107,15 +106,27 @@ class QLinearInt8 final : public torch::OperatorKernel {
     // This is the end of the pipeline, pass the resulting matrix through.
     fbgemm::DoNothing<> doNothingObj{};
 
+    at::Tensor bias = pack_ptr.bias;
+
     const int32_t* bias_ptr = nullptr;
-    if (bias.has_value()) {
-      Tensor bias_vec = bias.value();
-      TORCH_CHECK(bias_vec.dim() == 1, "bias should be a vector (1D Tensor)");
+
+    if (bias.defined()) {
+      // Temporary: Quantize bias
+      at::Tensor qbias = bias;
+      if (!bias.is_quantized()) {
+        qbias = at::quantize_linear(
+            bias, pack_ptr.w_scale[0] * input_scale_float, 0, kQInt32);
+      }
+      // FIXME Adding this causes ASAN issues in unit tests.
+      // else {
+      //   qbias = at::quantize_linear(
+      //       at::dequantize(bias), pack_ptr.w_scale[0] * input_scale_float, 0, kQInt32);
+      // }
+      TORCH_CHECK(qbias.dim() == 1, "bias should be a vector (1D Tensor)");
       TORCH_CHECK(
-          bias_vec.size(0) == N,
+          qbias.size(0) == N,
           "bias should have N elements: " + std::to_string(N));
-      // TODO: contiguous is called for further jit optimizations.
-      auto bias_contig = bias_vec.contiguous();
+      auto bias_contig = qbias.contiguous();
       bias_ptr =
```

Comment:
@supriyar @dzhulgakov 
We are grabbing a data pointer of a tensor in local scope here. Fixing this should fix the issue. I tried it locally and it worked.